modules/ts/src/ts_tags.cpp

```diff
@@ -280,6 +280,11 @@ static bool isTestTagSkipped(const std::string& testTag, CV_OUT std::string& ski
 
 void checkTestTags()
 {
+    if (std::find(skipped_tests.begin(), skipped_tests.end(),
+                  ::testing::UnitTest::GetInstance()->current_test_info()) != skipped_tests.end()) {
+        throw details::SkipTestExceptionBase(false);
```

Comment:
@dkurt Why do we need this?
What is the usecase? (usage example of the problem)

Reply:
Initially found in G-API tets where `findData` is used in tests constructor: https://github.com/dkurt/opencv/blob/609b045d8316166deae61908e0cbee87fb251a71/modules/gapi/test/infer/gapi_infer_ov_tests.cpp#L258-L259

Reply:
How does `findDataFile()` call `checkTestTags()`?

On failure `findDataFile()` throws `CV_Error()` or `SkipTestException` exception.

Reply:
You're right, `findDataFile` throws an exception but it is ignored on 4.x:
```cpp
TEST(TestFixture2, should_skip) {
    findDataFile("not_a_file.png", false);
}

struct TestFixture : public ::testing::Test {
    TestFixture() {
        printf("enter constructor\n");
        findDataFile("not_a_file.png", false);
        printf("finish constructor\n");
    }
};

TEST_F(TestFixture, Body) {
    printf("enter test body\n");
}
```

```
$ ./bin/opencv_test_core --gtest_filter=TestFixture*

Note: Google Test filter = TestFixture*
[==========] Running 2 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 1 test from TestFixture2
[ RUN      ] TestFixture2.should_skip
[     SKIP ] OpenCV tests: Can't find data file: not_a_file.png
[       OK ] TestFixture2.should_skip (0 ms)
[----------] 1 test from TestFixture2 (0 ms total)

[----------] 1 test from TestFixture
[ RUN      ] TestFixture.Body
enter constructor
unknown file: Failure
C++ exception with description "OpenCV tests: Can't find data file: not_a_file.png" thrown in the test fixture's constructor.
[  FAILED  ] TestFixture.Body (1 ms)
[----------] 1 test from TestFixture (1 ms total)

[----------] Global test environment tear-down
[ SKIPSTAT ] 2 tests skipped
[ SKIPSTAT ] TAG='skip_other' skip 2 tests
[==========] 2 tests from 2 test cases ran. (2 ms total)
[  PASSED  ] 1 test.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestFixture.Body
```

Reply:
In this PR `checkTestTags` is called for a dummy test object which returns by a factory if test fixture thrown SkipTestException.

Reply:
> `checkTestTags` is called for a dummy test object

Why not just `throw SkipTestException` instead?

Reply:
Were able to avoid this check for the case with exception at constructor. Exception at `SetUp` will probably require some isSkipped flag in `GTEST_TEST_CLASS_NAME_` (will do by a next commit).